cmake/BLTPrivateMacros.cmake

```diff
@@ -754,19 +754,63 @@ macro(blt_add_hip_executable)
 
 endmacro(blt_add_hip_executable)
 
+
+##-----------------------------------------------------------------------------
+## blt_make_file_ext_regex( EXTENSIONS   [ext1 [ext2 ...]]
+##                          OUTPUT_REGEX <regex variable name>)
+##
+## This function converts the list of extensions in EXTENSIONS and
+## fills the variable, given in OUTPUT_REGEX, with a joined, with '|',
+## regular expression. This regex should match any file name starting with
+## a string and ending with any one of the extensions in EXTENSIONS.
+## This also lower cases all extensions because we do not care about file casing.
+## -----------------------------------------------------------------------------
+macro(blt_make_file_ext_regex)
+
+    set(options)
+    set(singleValueArgs OUTPUT_REGEX)
+    set(multiValueArgs EXTENSIONS)
+
+    # Parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}"
+                            "${multiValueArgs}" ${ARGN} )
+
+    # Check arguments
+    if ( NOT DEFINED arg_EXTENSIONS )
+        message( FATAL_ERROR "Must provide a EXTENSIONS argument to the 'blt_make_file_ext_regex' macro" )
+    endif()
+
+    if ( NOT DEFINED arg_OUTPUT_REGEX )
+        message( FATAL_ERROR "Must provide a OUTPUT_REGEX argument to the 'blt_make_file_ext_regex' macro" )
+    endif()
+
+    # Join with 'or', and escape periods
+    list(JOIN arg_EXTENSIONS "|" ${arg_OUTPUT_REGEX})
+    # Lower-case because we do case-insensitive checks
+    string(TOLOWER "${BLT_C_FILE_REGEX}" BLT_C_FILE_REGEX)
+    # Escape periods before adding 
+    string(REPLACE "." "\\." ${arg_OUTPUT_REGEX} "${${arg_OUTPUT_REGEX}}")
+    # Regex for by any set of characters followed by any of the given
+    # file extensions at the end of the string
+    set(${arg_OUTPUT_REGEX} "^.*(${${arg_OUTPUT_REGEX}})$")
+
+endmacro(blt_make_file_ext_regex)
+
+
 ##------------------------------------------------------------------------------
 ## blt_split_source_list_by_language( SOURCES <sources>
 ##                                    C_LIST <list name>
 ##                                    Fortran_LIST <list name>
-##                                    Python_LIST <list name>)
+##                                    Python_LIST <list name>
 ##                                    CMAKE_LIST <list name>)
 ##
 ## Filters source list by file extension into C/C++, Fortran, Python, and
 ## CMake source lists based on BLT_C_FILE_EXTS, BLT_Fortran_FILE_EXTS,
-## and BLT_CMAKE_FILE_EXTS (global BLT variables). Files named
-## "CMakeLists.txt" are also filtered here. Files with no extension
-## or generator expressions that are not object libraries (of the form
-## "$<TARGET_OBJECTS:nameofobjectlibrary>") will throw fatal errors.
+## and BLT_CMAKE_FILE_EXTS (global BLT variables). This filtering is
+## case-insensitive. Files named "CMakeLists.txt" are also filtered here.
```

Comment:
Should we document somewhere that `BLT_{C,Fortran,CMAKE,Python}_FILE_EXTS` are global variables that can be overridden?

E.g. by adding a page to the user docs listing the variables that support being overridden? (And noting that doing so is an advanced feature?)

Reply:
They are documented here (https://llnl-blt.readthedocs.io/en/develop/api/code_check.html#blt-add-code-checks) but there could be a better place.

This goes along with previously discussed topic of documenting BLT's options better. Made an issue so we don't forget #547 and #548.